### ISSUE TITLE
Consider approachability and legibility of code

### ISSUE THREAD

**Issue by tomalrussell** [edwardoughton/itmlogic#45]
Capturing comments by @garrettj403 in the course of JOSS review.

Essential TODOs:
- [ ] [#43](https://github.com/edwardoughton/itmlogic/issues/43) comments in scripts
- [ ] [#44](https://github.com/edwardoughton/itmlogic/issues/44) docstrings in tests
- [x] fix [failing build](https://travis-ci.org/github/edwardoughton/itmlogic/builds/700223839)

Broader recommendation is to refactor and rename away from terse original variable names, and combine the subroutines into related modules and subpackages, with the goal of making the whole package a lot easier to read through and understand.

> I like that you provided two examples in the ``scripts/`` directory, but I think that they need a bit more explanation. For example, ``scripts/area.py`` doesn't have any inline comments past line 107, which makes it difficult to follow along. I raised this issue in [#43](https://github.com/edwardoughton/itmlogic/issues/43).
>
> The testing looks good (I think), but again it's very difficult to follow along without any comments. For example, I'm looking at the test for ``lrprop`` (``test_lrprop.py``). Where does ``setup_expected_prop_to_test_lrprop`` come from? Were these results generated by the previous FORTRAN/C++/MATLAB code? I think that you should have a short docstring in each file to describe the general idea behind the test. I raised this issue in [#44](https://github.com/edwardoughton/itmlogic/issues/44).
>
> Outside the scope of the JOSS review, I have a few general comments about the package:
>   - The variable names are very short, usually only 2 or 3 letters. Python is somewhat self-documenting, but that doesn't really work when you have short variable names. For example, you use ``gme`` to represent the Earth's effective curvature, but normally in Python you would call this variable something like ``earth_eff_curv``. I assume that this is because you adopted the variable names from the old C++ code.
>   - There are almost no inline comments, and since the variable names aren't obvious, it makes it very difficult to understand what's going on. I would suggest adding references. For example:
>      ```python
>      # Calculate attenuation
>      # Eqn. 6.3 in The ITS Irregular Terrain Model, the Algorithm - G. Hufford
>      ```
>   - The package is flat (no sub-directories) and each function is in its own separate module. To make it easier to navigate, you may want to consider grouping similar functions into modules and similar modules into sub-packages. Again, I assume that this is because you adopted the structure of some old MATLAB code.
>
> Overall, the package looks very useful. I think if you add a couple more comments to the ``scripts/`` and ``tests/`` directories, it will help new users to understand how the package works and ensure that the package is maintainable. 

_Originally posted by @garrettj403 in https://github.com/openjournals/joss-reviews/issues/2266#issuecomment-647231325_

**Comment by edwardoughton** [comment]
@garrettj403 I've proposed a new folder structure in the next iteration of corrections, which resembles the same function/module grouping as in the box diagram in the paper. Hopefully this will make things clearer.

**Comment by edwardoughton** [comment]
John (@garrettj403), we have now resolved:
- In-line comments throughout the runner scripts
- Doc string information in both modules and tests
- The flat structure (we now have a hierarchy which matches the diagram provided in the paper (for easy user reference)).

The only things we decided not to change currently were the variable names, as many users will be familiar with the names from the original code and we don't want to break from tradition in case it causes people not to use the repo. We decided this is something to consider in a future version, where a user may be able to specify if they want the traditional nomenclature or pythonic var names.
